PluginCode/src/main/java/smr/cs/ualberta/libcomp/action/EnableDomainAction.java

```diff
@@ -0,0 +1,317 @@
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.project.ProjectManager;
+import com.intellij.openapi.util.TextRange;
+import com.intellij.openapi.vfs.VirtualFile;
+import com.intellij.openapi.wm.WindowManager;
+import com.intellij.xdebugger.ui.DebuggerColors;
+import com.intellij.openapi.actionSystem.*;
+import com.intellij.openapi.command.WriteCommandAction;
+import com.intellij.openapi.editor.*;
+import com.intellij.psi.*;
+import com.intellij.psi.PsiImportList;
+import com.intellij.psi.PsiFile;
+import org.apache.commons.lang.StringUtils;
+import org.jetbrains.annotations.NotNull;
+import smr.cs.ualberta.libcomp.*;
+import smr.cs.ualberta.libcomp.data.DependencyStatement;
+import smr.cs.ualberta.libcomp.data.ReplacementFeedback;
+import smr.cs.ualberta.libcomp.data.ImportStatement;
+import smr.cs.ualberta.libcomp.data.User;
+import smr.cs.ualberta.libcomp.dialog.ReplacementDialog;
+
+import java.awt.*;
+import java.awt.event.WindowAdapter;
+import java.awt.event.WindowEvent;
+import java.io.IOException;
+import java.text.ParseException;
+import java.util.ArrayList;
+import java.util.Date;
+
+/**
+ * The ActionReplacement class is the main action for the plugin
+ * This is triggered by the replacement button on the main plugin dialog
+ */
+
+public class EnableDomainAction extends AnAction {
+
+    public int ExludedDomains;
+
+    public EnableDomainAction() {
+        ExludedDomains = 2;
+    }
+
+    @Override
+    public void actionPerformed(@NotNull final AnActionEvent event) {
+        ExludedDomains = 3;
+
+        Editor editor = event.getRequiredData(CommonDataKeys.EDITOR);
+        PsiFile psiFile = event.getRequiredData(CommonDataKeys.PSI_FILE);
+        final Project project = event.getRequiredData(CommonDataKeys.PROJECT);
+
+        if (psiFile != null) {
+            FileType fileType = psiFile.getFileType();
+
+            if (fileType.getDefaultExtension().equalsIgnoreCase("java")) {
+                detectImports(psiFile, editor, project);
+            }
+
+            if (fileType.getDefaultExtension().equalsIgnoreCase("groovy"))
+            {
+                try {
+                    detectDependancy(editor, psiFile , project);
+                } catch (IOException e) {
+                    e.printStackTrace();
+                }
+                event.getPresentation().setEnabledAndVisible(true);
+            }
+
+            if (fileType.getDefaultExtension().equalsIgnoreCase("xml"))
+            {
+                try {
+                    detectMaven(editor, psiFile , project);
+                } catch (IOException e) {
+                    e.printStackTrace();
+                }
+                event.getPresentation().setEnabledAndVisible(true);
+            }
+
+            event.getPresentation().setVisible(true);
+            event.getPresentation().setEnabled(true);
+        }
+    } // end of actionPErformed
+
+
+    public int detectDependenciesPSI(FileASTNode psinode)
```

Comment:
so when it says Dependency or Dependencies, does that mean Gradle? I see functions for maven but not gradle.. Should you rename these so it's clear it's for gradle?